halyard-config/src/main/java/com/netflix/spinnaker/halyard/config/spinnaker/v1/component/ClouddriverComponent.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright 2017 Google, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.halyard.config.spinnaker.v1.component;
+
+import com.netflix.spinnaker.halyard.config.model.v1.node.DeploymentConfiguration;
+import org.springframework.stereotype.Component;
+
+@Component
+public class ClouddriverComponent extends SpringComponent {
+  @Override
+  public String getComponentName() {
+    return "clouddriver";
+  }
+
+  @Override
+  public String generateFullConfig(String baseConfig, DeploymentConfiguration deploymentConfiguration) {
+    return yamlToString(deploymentConfiguration.getProviders()) + "\n" + baseConfig;
```

Comment:
The set of key/value pairs in `deploymentConfiguration.getProviders()` doesn't have any intersection with what's in `baseConfig`, right? Otherwise I think you could overwrite keys in the latter overwrite the former.